### ISSUE TITLE
Feature: select output destination based on message fields

### ISSUE THREAD

**Issue by manuelluis** [fluent/fluent-bit#812]
**Is your feature request related to a problem? Please describe.**

If we have one input, for example systemd, and two outputs and we cannot filter the output based on a tag, and we need to send all data to one of the output but only some data to the other output based on some input field.                                                                                                    

**Describe the solution you'd like**

Allow to rewrite tags in filters.

For example in the lua filter we can return the tag.

```
function cb_print(tag, timestamp, record)
   return code, tag, timestamp, record
end
```

or without breaking the existing configurations, put at the end and make optional
```
function cb_print(tag, timestamp, record)
   return code, timestamp, record, tag
end
```

**Describe alternatives you've considered**

Duplicate the inputs in the configuration, but in some cases can cause a high overhead

**Additional context**

**Comment by manuelluis** [comment]
Since de tags are for a groups of messages, if the filter can change the tag of a message it will implied a lot of big changes, maybe put the tag at message level (tag, timestamp, message) an change all plugins.

Maybe it will more simple a filter that duplicate messages with a different tag, It should require less changes

```
[FILTER]
    Name   dup
    Match  *
    Tag    new-tag
```

@edsiper Would this be acceptable?

**Comment by nokute78** [comment]
Fluentd doesn’t allow rewriting tag at filter by design.
It makes dataflow complex.
(see also #291 I tried to do that )

I think what we need is “label” feature of fluentd.

https://docs.fluentd.org/v1.0/articles/life-of-a-fluentd-event#labels
https://docs.fluentd.org/v1.0/articles/routing-examples#re-route-event-to-other-label

**Comment by edsiper** [comment]
New rewrite_tag has landed in GIT master, more details on #293
